app/controllers/cart_controller.rb

```diff
@@ -0,0 +1,30 @@
+class CartController < ApplicationController
+  load_and_authorize_resource class: Order
+  before_action :set_order
+
+  def show; end
+
+  def update
+    @carts_update = Carts::UpdatePresenter.new(current_user_order, params[:code])
```

Comment:
please use more obvious naming e.g.
`@carts_update => @presenter`
`Carts::UpdatePresenter => Catrs::Presenters::Update or Presenters::Carts::Update`

Reply:
Also it seems you misunderstood the purpose of Presenters
Presenters shouldn't contain any business logic
basically Presenters and Decorators are the same, but is used for different cases
https://stackoverflow.com/questions/7860301/ruby-on-rails-patterns-decorator-vs-presenter
Please move business logic for coupon to service object

Reply:
Also fix other presenters if they contain business logic